File: leetcode/1047.remove-all-adjacent-duplicates-in-string/cpp/removeduplicates_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "removeduplicates.cpp"

struct testCase
{
    std::string s;
    std::string want;
};

TEST_CASE("IsValid")
{
    std::vector<testCase>
        tcs{
            {"abbaca", "ca"},
            {"azxxzy", "ay"},
        };

    Solution s;

    for (auto tc : tcs)
    {
        auto got = s.removeDuplicates(tc.s);
        CHECK(got == tc.want);
    }
}

File: leetcode/1047.remove-all-adjacent-duplicates-in-string/ts/removeduplicates.test.ts
import { test, expect } from "vitest"
import removeDuplicates from "./removeduplicates.js"

interface testcase {
    s: string
    want: string
}

test("removeduplicates", () => {
    const tcs: testcase[] = [
        { s: "abbaca", want: "ca" },
        { s: "azxxzy", want: "ay" }
    ]

    tcs.forEach(tc => expect(removeDuplicates(tc.s) === tc.want))
})
